andela/ah-orcas
# #161254663 create, list, update and delete articles

#### What does this PR do?
Displays the following added functionalities:
- user can view articles
- user can create articles
- user can update articles
- user can delete articles

#### Description of Task to be completed?
- Articles form the backbone of our application and introduce us to CRUD which is the building block of most web applications.

#### How should this be manually tested?
- Clone this repo
```code
$ git clone https://github.com/andela/ah-orcas
```
- Navigate to this branch
```code
$ git checkout ft-create-articles-161254663
```
- Setup the environment variables, database and install the requirements. See sample.env
```code
$ source .env
```
- Start the server 
```code
$ python manage.py runserver
```
- Hit the endpoints created.
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/4da362e65fb40434f1d4)

#### Any background context you want to provide?
- I have rebased with Develop and as a result I found an existing folder called articles which I have successfully merged together with my work. I have however renamed the folder to article. All the work from articles folder has been integrated to the new folder article. 

#### What are the relevant pivotal tracker stories?
[161254663](https://www.pivotaltracker.com/story/show/161254663)

#### Screenshots (if appropriate)
- Get article endpoint screenshot
<img width="1078" alt="screen shot 2018-11-08 at 12 20 20" src="https://user-images.githubusercontent.com/26874561/48189282-e09bdd00-e350-11e8-874a-157d7570f3b1.png">

- Create article endpoint screenshot 
<img width="1036" alt="screen shot 2018-11-08 at 12 20 36" src="https://user-images.githubusercontent.com/26874561/48189323-f6a99d80-e350-11e8-9863-846ba92b0d56.png">

- Update article endpoint screenshot
<img width="1073" alt="screen shot 2018-11-08 at 12 20 45" src="https://user-images.githubusercontent.com/26874561/48189347-075a1380-e351-11e8-955c-3f2bc91ff323.png">

- Delete article endpoint screenshot
<img width="1058" alt="screen shot 2018-11-08 at 12 20 58" src="https://user-images.githubusercontent.com/26874561/48189368-12ad3f00-e351-11e8-9316-d4d5456637c4.png">

#### Questions:
- What do you think can be improved on?
